plugins/accelerated-peft/src/fms_acceleration_peft/autogptq_utils.py

```diff
@@ -32,6 +32,18 @@
 # these parameters are to be patched for triton v2
 # consider making a map if patching more kernels
 PATCH_FOR_FSDP_TRITON_V2 = ["qweight", "qzeros"]
+PEFT_ALL_LINEAR = "all-linear"
+
+
+def requires_installation_on_all_linears(peft_config):
+    tm = peft_config.target_modules
+    assert isinstance(tm, (list, str)), "target modules can only be list or string"
+    if isinstance(tm, list):
+        if PEFT_ALL_LINEAR not in tm:
+            return False
+        assert len(tm) == 1, f"`{PEFT_ALL_LINEAR}` must exist alone in target modules"
```

Comment:
ahh interesting. So we don't allow someone passing `target_modules` as `["all_linear", "lm_head"]`? We allow this in our doc for LoRA as HF allows it for loRA https://github.com/foundation-model-stack/fms-hf-tuning?tab=readme-ov-file#lora-tuning-example (see the section "How to specify lm_head as a target module", example 3). I can update the doc however for qLoRA. Or should we keep consistent with loRA and allow both values?

Up to you! I will approve the PR so it's not blocked on me for merging for this release, but we can open new PRs to amend this behavior

Reply:
@aluu317 in your doc, I understand `["all_linear", "lm_head"]` as equivalent to `["all_linear"]`, is that correct? That is, `all-linear` by default includes the `lm_head` also?

Reply:
no, what that doc means is running `["all_linear", "lm_head"]` is same as running `["all_linear"]` only. `lm_head` will not be produced in either way

Reply:
Sorry I am not clear, are you saying `["all_linear", "lm_head"]` will not install adapters onto `lm_head`?